glib/src/param_spec.rs

```diff
@@ -1315,16 +1646,67 @@ impl ParamSpecVariant {
             }
         }
     }
+
+    pub fn builder<'a>(name: &'a str, type_: &'a crate::VariantTy) -> ParamSpecVariantBuilder<'a> {
+        ParamSpecVariantBuilder::new(name, type_)
+    }
+}
+
+#[must_use]
+pub struct ParamSpecVariantBuilder<'a> {
+    name: &'a str,
+    nick: Option<&'a str>,
+    blurb: Option<&'a str>,
+    flags: crate::ParamFlags,
+    type_: &'a crate::VariantTy,
+    default_value: Option<&'a crate::Variant>,
+}
+
+impl<'a> ParamSpecVariantBuilder<'a> {
+    fn new(name: &'a str, type_: &'a crate::VariantTy) -> Self {
+        Self {
+            name,
+            nick: None,
+            blurb: None,
+            flags: crate::ParamFlags::default(),
+            type_,
+            default_value: None,
+        }
+    }
+
+    #[doc = "Default: None`"]
+    pub fn default_value(mut self, value: impl Into<Option<&'a crate::Variant>>) -> Self {
+        self.default_value = value.into();
+        self
+    }
+
+    #[must_use]
+    pub fn build(self) -> ParamSpec {
+        ParamSpecVariant::new(
+            self.name,
+            self.nick,
+            self.blurb,
+            self.type_,
+            self.default_value,
+            self.flags,
+        )
+    }
 }
 
-define_builder!(
-    ParamSpecVariant,
-    ParamSpecVariantBuilder {
-        type_: &'a crate::VariantTy,
-        default_value: Option<&'a crate::Variant> = None,
+impl<'a> crate::prelude::ParamSpecBuilderExt<'a> for ParamSpecVariantBuilder<'a> {
+    fn set_nick(&mut self, nick: Option<&'a str>) {
+        self.nick = nick;
     }
-    requires (type_: &'a crate::VariantTy,)
-);
+    fn set_blurb(&mut self, blurb: Option<&'a str>) {
+        self.blurb = blurb;
+    }
+    fn set_flags(&mut self, flags: crate::ParamFlags) {
+        self.flags = flags;
+    }
+    fn current_flags(&self) -> crate::ParamFlags {
+        self.flags
+    }
+}
```

Comment:
completely random but maybe we can make this trait implemented by a small macro if you want ? :)

Reply:
Not so easy because some builders are `<'a>`, others are `<'a, T>` etc

Reply:
I'd keep it like this for now.